cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -126,45 +155,121 @@
     this.streamCoordinatorClient = streamCoordinatorClient;
     this.privilegesManager = privilegesManager;
     this.authorizationEnforcer = authorizationEnforcer;
+    // TODO make maximum size and expire after write configurable?
+    this.appInjectors = CacheBuilder.newBuilder()
```

Comment:
the cache is so you don't create a new injector every time preview is run with the same configuration? Should the injector be different for two apps with the same ``ApplicationId`` but different ``AppRequest``s?